src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -49,6 +60,12 @@ await Task.WhenAny(_flushTask, Task.Delay(TimeSpan.FromSeconds(20)))
         private async Task FlushTracesAsync()
         {
             var traces = _tracesBuffer.Pop();
+            var spanCount = traces.Sum(t => t.Count);
+
+            _dogStatsdClient?.Gauge(TracerMetricNames.Queue.DequeuedTraces, traces.Count);
+            _dogStatsdClient?.Gauge(TracerMetricNames.Queue.DequeuedSpans, spanCount);
+            _dogStatsdClient?.Gauge(TracerMetricNames.Queue.TraceQueueMaxCapacity, TraceBufferSize);
```

Comment:
Do we need to send this every single time? Won't it always be 1k? Or at least the same for the life of one process.

Reply:
If this section is batched, I don't really care how many you send up.

Reply:
Its batches now, but to avert the original question, we do need to keep sending once in a while to keep the metric "alive." If stop sending it it won't show up in the UI after a while.